scripts/cloudbuild.py

```diff
@@ -0,0 +1,179 @@
+#!/usr/bin/env python
+'''creates cloudbuild.json for caliban'''
+
+from absl import flags, app, logging
+from typing import Dict, List, Any, NamedTuple, Optional
+
+import json
+import pprint as pp
+
+# ----------------------------------------------------------------------------
+_CLOUD_BUILDER = 'gcr.io/cloud-builders/docker'
+_BASE_URL = 'gcr.io/blueshift-playground/blueshift'
+_GPU_BASE_TAG = 'base'
+_CPU_TAG = 'cpu'
+_GPU_TAG = 'gpu'
+_CPU_BASE_IMAGE = 'ubuntu:18.04'
+_CPU_DOCKERFILE = 'dockerfiles/Dockerfile'
+_GPU_DOCKERFILE = 'dockerfiles/Dockerfile.gpu'
+
+# ----------------------------------------------------------------------------
+FLAGS = flags.FLAGS
+
+flags.DEFINE_string('output', None, 'output json file')
+flags.DEFINE_string('config', None, 'input configuration file')
+flags.DEFINE_string('base_url', _BASE_URL, 'base url for images')
+flags.DEFINE_integer('timeout_sec', 24 * 60 * 60,
+                     'timeout for all builds to complete')
+
+# ----------------------------------------------------------------------------
+logging.get_absl_handler().setFormatter(None)
+
+Config = Dict[str, Any]
+
+
+# ----------------------------------------------------------------------------
+class BuildStep(NamedTuple):
+  '''a google cloud-build step'''
+  sid: str
```

Comment:
is this `step_id`? can we call it `id`, or `step_id`?

Reply:
Agree, that's cleaner. I'll change that.